components/brave_account/brave_account_service.cc

```diff
@@ -387,4 +403,127 @@ void BraveAccountService::OnVerifyResult(VerifyResult::Response response) {
   ScheduleVerifyResult(kVerifyResultPollInterval);
 }
 
+void BraveAccountService::OnLoginInitialize(LoginInitializeCallback callback,
+                                            LoginInit::Response response) {
+  if (!response.body) {
+    return std::move(callback).Run(base::unexpected(mojom::LoginError::New(
+        response.status_code.value_or(response.net_error), std::nullopt)));
+  }
+
+  CHECK(response.status_code);
+  const auto status_code = *response.status_code;
+
+  auto result =
+      std::move(*response.body)
+          // expected<SuccessBody, [ErrorBody ]> ==>
+          // expected<SuccessBody, [LoginError]>
+          .transform_error([&](auto error_body) {
+            return mojom::LoginError::New(
+                status_code,
+                TransformError<mojom::LoginErrorCode>(std::move(error_body)));
+          })
+          // expected<[SuccessBody             ], LoginError> ==>
+          // expected<[LoginInitializeResultPtr], LoginError>
+          .and_then([&](auto success_body)
+                        -> base::expected<mojom::LoginInitializeResultPtr,
+                                          mojom::LoginErrorPtr> {
+            if (success_body.login_token.empty() ||
+                success_body.serialized_ke2.empty()) {
+              return base::unexpected(
+                  mojom::LoginError::New(status_code, std::nullopt));
+            }
+
+            std::string encrypted_login_token =
+                Encrypt(success_body.login_token);
+            if (encrypted_login_token.empty()) {
+              return base::unexpected(mojom::LoginError::New(
+                  std::nullopt,
+                  mojom::LoginErrorCode::kLoginTokenEncryptionFailed));
+            }
+
+            return mojom::LoginInitializeResult::New(
+                std::move(encrypted_login_token),
```

Comment:
does it make sense for the webui to have the login token? Perhaps we should just cache it in the service somehow

Reply:
In the current design I do pass the login token back to the WebUI, even though it does not strictly need it. The idea was to keep the OPAQUE flow self-contained on the frontend for now, and only use encryption here so we are not handing the token back to JS in plaintext.

Longer term, the plan is to move this orchestration into the browser side. I will revisit this as part of https://github.com/brave/brave-browser/issues/49350, where I'm planning to store the login token (and other state) in the context of the `ReceiverSet<>` inside `BraveAccountService` instead of bouncing it through the WebUI.

Reply:
> I can't quite guarantee that yet

> BraveAccountService is shared across all receivers, but the login token would be tied to a specific WebUI flow

That's okay, I'm only suggesting simplifying things to act as a placeholder for what's to come. Not a blocker imo, so up to you.